File: packages/sdk/src/logger.ts
export class Logger {
  private enabled: boolean;

  constructor(enabled: boolean) {
    this.enabled = enabled;
  }

  debug(message: string, ...args: unknown[]): void {
    if (this.enabled) {
      console.debug(`[EdgeFlags] ${message}`, ...args);
    }
  }

  warn(message: string, ...args: unknown[]): void {
    if (this.enabled) {
      console.warn(`[EdgeFlags] ${message}`, ...args);
    }
  }

  error(message: string, ...args: unknown[]): void {
    if (this.enabled) {
      console.error(`[EdgeFlags] ${message}`, ...args);
    }
  }
}

File: packages/python/src/edgeflags/logger.py
from __future__ import annotations

import logging

_log = logging.getLogger("edgeflags")


class Logger:
    def __init__(self, enabled: bool) -> None:
        self._enabled = enabled

    def debug(self, message: str, *args: object) -> None:
        if self._enabled:
            _log.debug("[EdgeFlags] %s", message, *args)

    def warn(self, message: str, *args: object) -> None:
        if self._enabled:
            _log.warning("[EdgeFlags] %s", message, *args)

    def error(self, message: str, *args: object) -> None:
        if self._enabled:
            _log.error("[EdgeFlags] %s", message, *args)
